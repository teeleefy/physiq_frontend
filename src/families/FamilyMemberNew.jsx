import { Form, FormGroup, Input, Button, Label, Alert } from "reactstrap";
import { useState, useContext } from "react";
import {FamilyContext} from "../auth/UserContext.js";
import PhysiqApi from "../Api.js";
import { useNavigate } from "react-router-dom";
import './styles/FamilyMember.css'
import { toast } from 'react-toastify';

function FamilyMemberNew(){
    let navigate = useNavigate();
    const { currentFamily } = useContext(FamilyContext);
    // console.log(currentFamily);
    const [formMessages, setFormMessages] = useState([]);
    const [updateSuccess, setUpdateSuccess] =useState();
    const [formData, setFormData] = useState({firstName: "", lastName: "", birthday: ""});

    async function addFamilyMember() {
        try {
          let newFamilyMember= await PhysiqApi.addFamilyMember(currentFamily.id, formData);
          return { success: true };
        } catch (errors) {
          console.error("Add FamilyMember failed", errors);
          return { success: false, errors };
        }
      }

    async function handleSubmit(evt){
        evt.preventDefault();
        // console.log(formData);
        let result = await addFamilyMember();
        
        if(result.success){
            toast.success('Member Added!', {
                position: "top-center",
                autoClose: 2000,
                hideProgressBar: false,
                closeOnClick: true,
                pauseOnHover: true,
                draggable: true,
                progress: undefined,
                theme: "dark",
                }); 
           setFormMessages(['Family Member Added!'])
           setUpdateSuccess(true);
           navigate("/home");
        }
        else{
            setFormMessages(result.errors);
            setUpdateSuccess(false);
        }
    };

  
    /** Update local state w/curr state of input elem */
  
    const handleChange = evt => {
      const { name, value }= evt.target;
      setFormData(fData => ({
        ...fData,
        [name]: value
      }));
    };

    return(
        <div id="FamilyMember-main">
            <Form className="FamilyMember-Form m-4" >
            <h1 className="FamilyMember-h1">Add Family Member:</h1>
            <p className="text-secondary">Fields marked with <span className="text-danger">*</span> are required.</p>
            <FormGroup>
                <Label className="FamilyMember-label" for="firstName" >
                    <b>First Name</b><span className="text-danger">*</span>
                </Label>
                <Input
                id="firstName"
                name="firstName"
                minLength={2}
                maxLength={35}
                value={formData.firstName}
                placeholder="Enter first name..."
                type="text"
                onChange={handleChange}
                />
            </FormGroup>
            <FormGroup>
                <Label className="FamilyMember-label" for="lastName" >
                    <b>Last Name</b><span className="text-danger">*</span>
                </Label>
                <Input
                id="lastName"
                name="lastName"
                minLength={2}
                maxLength={50}
                value={formData.lastName}
                placeholder="Enter last name..."
                type="text"
                onChange={handleChange}
                />
            </FormGroup>
           
            <FormGroup className="FamilyMember-date-input">
                <Label className="FamilyMember-label" for="birthday">
                    <b>Birthday</b>
                </Label>
                <Input
                id="birthday"
                name="birthday"
                value={formData.birthday}
                placeholder="Enter birthday..."
                type="date"
                onChange={handleChange}
                />
            </FormGroup>
                {formMessages.length
                    ? formMessages.map(msg => <Alert color={updateSuccess ? "success": "danger"}>{msg}</Alert>)
                    : null
                }
                
                <Button className="btn-dark" onClick={handleSubmit}>Add Family Member</Button>
            </Form>
        </div>
    )}
    
export default FamilyMemberNew;