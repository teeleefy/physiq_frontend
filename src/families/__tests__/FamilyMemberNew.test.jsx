import {renderWithRouterAndFamilyContext, screen} from '../../../__tests__/test-tools';
import FamilyMemberNew from '../FamilyMemberNew';

describe('FamilyMemberNew', () => {
    it('renders the FamilyMemberNew component', () => {
      renderWithRouterAndFamilyContext(<FamilyMemberNew/>);
      screen.debug(); // prints out the jsx in the App component unto the command line
    });

    it("matches snapshot", function() {
      const { asFragment } = renderWithRouterAndFamilyContext(<FamilyMemberNew />);
      expect(asFragment()).toMatchSnapshot();
    });

  });
