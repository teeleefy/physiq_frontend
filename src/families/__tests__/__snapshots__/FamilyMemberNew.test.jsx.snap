// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FamilyMemberNew > matches snapshot 1`] = `
<DocumentFragment>
  <div
    id="FamilyMember-main"
  >
    <form
      class="FamilyMember-Form m-4"
    >
      <h1
        class="FamilyMember-h1"
      >
        Add Family Member:
      </h1>
      <p
        class="text-secondary"
      >
        Fields marked with 
        <span
          class="text-danger"
        >
          *
        </span>
         are required.
      </p>
      <div
        class="mb-3"
      >
        <label
          class="FamilyMember-label form-label"
          for="firstName"
        >
          <b>
            First Name
          </b>
          <span
            class="text-danger"
          >
            *
          </span>
        </label>
        <input
          class="form-control"
          id="firstName"
          maxlength="35"
          minlength="2"
          name="firstName"
          placeholder="Enter first name..."
          type="text"
          value=""
        />
      </div>
      <div
        class="mb-3"
      >
        <label
          class="FamilyMember-label form-label"
          for="lastName"
        >
          <b>
            Last Name
          </b>
          <span
            class="text-danger"
          >
            *
          </span>
        </label>
        <input
          class="form-control"
          id="lastName"
          maxlength="50"
          minlength="2"
          name="lastName"
          placeholder="Enter last name..."
          type="text"
          value=""
        />
      </div>
      <div
        class="FamilyMember-date-input mb-3"
      >
        <label
          class="FamilyMember-label form-label"
          for="birthday"
        >
          <b>
            Birthday
          </b>
        </label>
        <input
          class="form-control"
          id="birthday"
          name="birthday"
          placeholder="Enter birthday..."
          type="date"
          value=""
        />
      </div>
      <button
        class="btn-dark btn btn-secondary"
        type="button"
      >
        Add Family Member
      </button>
    </form>
  </div>
</DocumentFragment>
`;
